server.js

```diff
@@ -1,32 +1,58 @@
 const bodyParser = require('body-parser');
 const express = require('express');
 const mongoose = require('mongoose');
+const Bear = require('./models.js');
 
 const STATUS_USER_ERROR = 422;
 const STATUS_SERVER_ERROR = 500;
 const server = express();
 
-// allow server to parse JSON bodies from POST/PUT/DELETE requests
 server.use(bodyParser.json());
 
-
-
-
-
-
-// TODO: write your server code here
-
-
-
-
-
-
 mongoose.Promise = global.Promise;
 const connect = mongoose.connect(
   'mongodb://localhost/bears',
   { useMongoClient: true }
 );
 
+server.get('/bears', (req, res) => {
+  Bear.find({}, (err, bears) => {
+    if (err) {
+      res.status(STATUS_SERVER_ERROR);
+      res.json(err);
+    }
+    res.json(bears);
+  });
+});
+
+server.get('/bears/:id', (req, res) => {
+  const { id } = req.params;
+  Bear.findById(id, (err, bear) => {
+    if (err) {
+      res.status(STATUS_SERVER_ERROR);
+      res.json(err);
+    }
+    res.json(bear);
+  });
+});
+
+server.post('/bears', (req, res) => {
+  const { species, latinName } = req.body;
+  if (!species || !latinName) {
+    res.status(STATUS_USER_ERROR);
+    res.json({ error: 'Please provide the bear\'s species and Latin name' });
```

Comment:
Awesome error handling! I'm glad this wasn't lost in my solution lecture. This is very good practice for sure!